File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02734/p02734_num7_parsed.cpp
#include <bits/stdc++.h>
using namespace std;

const int mod = 998244353;

int main() {
    // Read the input values
    int N, S;
    cin >> N >> S;
    vector<int> A(N);
    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }
    
    // Initialize the dp array
    vector<vector<int>> dp(N, vector<int>(S + 1, 0));
    
    // Base case
    if (A[0] <= S) {
        dp[0][A[0]] = 1;
    }
    
    // Dynamic programming
    for (int i = 1; i < N; i++) {
        for (int j = 0; j <= S; j++) {
            dp[i][j] = (dp[i][j] + dp[i - 1][j]) % mod;
            if (j >= A[i]) {
                dp[i][j] = (dp[i][j] + dp[i - 1][j - A[i]]) % mod;
            }
        }
    }
    
    // Calculate the sum of f(L, R)
    int sum = 0;
    for (int i = 0; i < N; i++) {
        for (int j = i; j < N; j++) {
            sum = (sum + dp[j][S]) % mod;
        }
    }
    
    // Print the result
    cout << sum << endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02734/p02734_num7_parsed.py
import sys

MOD = 998244353

def sum_f_lr(n, s, a):
    dp = [[0] * (s + 1) for _ in range(n + 1)]
    dp[0][0] = 1

    for i in range(1, n + 1):
        for j in range(s + 1):
            dp[i][j] = (2 * dp[i - 1][j]) % MOD
            if j >= a[i - 1]:
                dp[i][j] = (dp[i][j] + dp[i - 1][j - a[i - 1]]) % MOD

    return sum(dp[n]) % MOD

def main():
    n, s = map(int, input().split())
    a = list(map(int, input().split()))

    result = sum_f_lr(n, s, a)
    print(result)

if __name__ == "__main__":
    main()